### PR TITLE
chore: Make font-weights themable, fix font faces

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
1. Due to incorrect import of fonts in `src/assets/stylesheets/less/fonts.less`, in fact only the fonts with normal font weight were available. That means that font weights other than 400 simply didn't work - weights from 200 to 500 were rendered as 400, bold font weights were extrapolated by the browser. This PR fixes that issue by explicitely importing font files with weights 200, 400, 500 and 600.
2. After the change described above, font-weight 700 (bold) was too thick. I replaced all 700 weights with 600, so that all texts look similar to how they looked before.
3. HTML tags `<h1-6>`, `<th>` and `<strong>` all use `font-weight: 700` by default. I've overriden them globally to use `font-weight: 600`.
4. I changed the value of bold font weight in `supersetTheme` from 700 to 600, because of the reasons described above.
5. I replaced hardcoded font weights in many components with superset theme variables. Only a few font weights in some plugins css files remain hardcoded.

### TESTING INSTRUCTIONS
The changes affect the whole product - everything should look as it did before (if you compare side-by-side with current look, there are slight differences in how bold texts look - that's because we didn't actually use Inter font's bold version before).

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

CC @kasiazjc

### CODE REVIEW

**Comment by michael-s-molina** [comment]
@jinghua-qa Can you please help us test this one?

File: superset-frontend/packages/superset-ui-core/test/chart/components/SuperChart.test.tsx [hunk 1]
```diff
@@ -18,12 +18,14 @@
...
 
 import React from 'react';
-import { mount } from 'enzyme';
 import mockConsole, { RestoreConsole } from 'jest-mock-console';
 import { triggerResizeObserver } from 'resize-observer-polyfill';
 import ErrorBoundary from 'react-error-boundary';
 
 import { promiseTimeout, SuperChart } from '@superset-ui/core';
+// @ts-ignore
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (michael-s-molina) [inline superset-frontend/packages/superset-ui-core/test/chart/components/SuperChart.test.tsx:27]: I'm not sure if we should create this dependency. Especially because we're moving the files in `spec` to `src`. Maybe let's just copy `styledMount` logic for now while we think about cyclic dependencies.

File: superset-frontend/plugins/plugin-chart-pivot-table/src/react-pivottable/Styles.js [hunk 2]
```diff
@@ -17,123 +17,124 @@
...
-    cursor: pointer;
-  }
+  ${({ theme }) => css`
+    table.pvtTable {
+      position: relative;
+      font-size: ${theme.typography.sizes.s}px;
+      text-align: left;
+      margin: ${theme.gridUnit}px;
+      border-collapse: separate;
+      font-family: 'Inter', Helvetica, Arial, sans-serif;
```
Comment (michael-s-molina) [inline superset-frontend/plugins/plugin-chart-pivot-table/src/react-pivottable/Styles.js:30]:
```suggestion
      font-family: ${theme.typography.families.sansSerif};
```

File: superset-frontend/src/GlobalStyles.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { css } from '@superset-ui/core';
+import { Global } from '@emotion/react';
+
+export const GlobalStyles = () => (
```
Comment (michael-s-molina) [inline superset-frontend/src/GlobalStyles.tsx:23]: If you search for `<Global` you'll find only 8 files. It would be a BIG plus to move these styles to this file and create a linting rule that prevents any `Global` declaration outside this file. You can do this in another PR if you think it's out of scope.

File: superset-frontend/src/dashboard/components/FiltersBadge/Styles.tsx [hunk 4]
```diff
@@ -16,68 +16,70 @@
...
-    color: ${({ theme }) => theme.colors.grayscale.light5};
-    width: 1em;
-    height: 1em;
+  ${({ theme }) => css`
     display: inline-block;
+    color: ${theme.colors.grayscale.light5};
+    background: ${theme.colors.grayscale.base};
+    border-radius: 1em;
+    vertical-align: text-top;
+    padding: ${`${theme.gridUnit}px ${theme.gridUnit * 2}px`};
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/FiltersBadge/Styles.tsx:28]:
```suggestion
    padding: ${theme.gridUnit}px ${theme.gridUnit * 2}px;
```

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/styles.ts [hunk 5]
```diff
@@ -284,85 +280,86 @@ export const validatedFormStyles = (theme: SupersetTheme) => css`
...
+    &.mb-8 {
+      margin-bottom: ${theme.gridUnit * 2}px;
+    }
 
-  &.extra-container {
-    padding-top: 8px;
-  }
+    .control-label {
+      color: ${theme.colors.grayscale.dark1};
+      font-size: ${theme.typography.sizes.s - 1}px;
```
Comment (michael-s-molina) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/styles.ts:294]: Do we really need the `-1`? Can't we just change it to `s`?

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/styles.ts [hunk 6]
```diff
@@ -284,85 +280,86 @@ export const validatedFormStyles = (theme: SupersetTheme) => css`
...
+      margin-bottom: ${theme.gridUnit * 2}px;
+    }
 
-  .input-container {
-    display: flex;
-    align-items: top;
+    &.extra-container {
+      padding-top: 8px;
```
Comment (michael-s-molina) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/styles.ts:299]:
```suggestion
      padding-top: ${theme.gridUnit * 2}px;
```

File: superset-frontend/src/dashboard/components/FiltersBadge/Styles.tsx (hunk 4 above)
Comment (kgabryje) [reply superset-frontend/src/dashboard/components/FiltersBadge/Styles.tsx:28]: 🤦

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/styles.ts (hunk 5 above)
Comment (kgabryje) [reply superset-frontend/src/views/CRUD/data/database/DatabaseModal/styles.ts:294]: I really don't like "hacking" our theme like that, but I'm kinda nervous about making such changes in this PR... It's not obvious if it should be `s` or `xs`.

File: superset-frontend/packages/superset-ui-core/src/chart/components/NoResultsComponent.tsx [hunk 7]
```diff
@@ -17,31 +17,38 @@
...
+}>`
+  ${({ theme, width, height }) => css`
+    align-items: center;
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    text-align: center;
+    height: ${height}px;
+    width: ${width}px;
+    //padding: ${theme.gridUnit * 4}px;
```
[the remark this answers is not in the document]
Comment (kgabryje) [reply superset-frontend/packages/superset-ui-core/src/chart/components/NoResultsComponent.tsx]: Commented by mistake

**Review by michael-s-molina — APPROVED** [review]
LGTM. Thanks for the improvements!
